PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/arm/ResourceTemplateView.java

```diff
@@ -102,16 +101,20 @@ public void beforeFileClosed(@NotNull FileEditorManager source, @NotNull Virtual
                 if (file.getFileType().getName().equals(ResourceTemplateViewProvider.TYPE) &&
                         file.getName().equals(node.getName())) {
                     try {
-                        if(isTemplateUpdate()){
-                            if (DefaultLoader.getUIHelper().showConfirmation(PROMPT_MESSAGE_SAVE_TEMPALTE, "Azure Explorer",
-                                    new String[]{"Yes", "No"}, null)) {
-                                new ExportTemplate(node).doExport(getTemplate());
-                            }
+                        boolean templateUpdate = isTemplateUpdate();
+                        boolean propertiesUpdate = isPropertiesUpdate();
+                        if (!templateUpdate && !propertiesUpdate) {
+                            return;
                         }
-                        if(isPropertiesUpdate()){
-                            if (DefaultLoader.getUIHelper().showConfirmation(PROMPT_MESSAGE_SAVE_PARAMETERS, "Azure Explorer",
-                                    new String[]{"Yes", "No"}, null)) {
-                                new ExportTemplate(node).doExportParameters(getParameters());
+                        if (DefaultLoader.getUIHelper().showConfirmation(PROMPT_MESSAGE_SAVE, "Azure Explorer",
```

Comment:
Duplicate code of show yes/no confirmation, make it in an util class

Reply:
Done.